src/App.js

```diff
@@ -26,40 +27,122 @@ const generateSquares = () => {
 };
 
 const App = () => {
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
+  // initialize state
   const [squares, setSquares] = useState(generateSquares());
 
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+
+  const [winnerLabel, setWinnerLabel] = useState('');
+
   // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
-
-  const checkForWinner = () => {
-    // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if
-    //    all three squares have the same value.
+  const clickSquare = (mySquare) => {
+    for (let i = 0; i < 3; i++) {
+      for (let n = 0; n < 3; n++) 
+        if (squares[i][n].id == mySquare) {
+          if (squares[i][n].value == '') {
+            squares[i][n].value = currentPlayer;
+          } else {
+            return false;
+          }
+        }
+      }
+    setSquares([...squares]);
+
+    //Now that we put the mark on the square, check and see if they won by calling checkForWinner()
+    let isWon = checkForWinnerChickenDinner();
```

Comment:
One note.  the setState function `setSquares` is asynchronous.  It will change the state... eventually (within a few miliseconds).  The reason your code works is that you manually changed the state variable above.